crates/oxc_parser/src/lexer/mod.rs

```diff
@@ -115,7 +114,7 @@ impl<'a> Lexer<'a> {
             source_type,
             token,
             errors: vec![],
-            lookahead: VecDeque::with_capacity(4), // 4 is the maximum lookahead for TypeScript
```

Comment:
looks like this comment is wrong, while 4 maybe the max lookahead in the general case code such as
```
type x = {
  static static static static static static static foo: string
}
```
causes the assertion to fail

Reply:
Is that why you closed this PR? I've wanted to do exactly what you've done here for a while, and I noticed the problem with modifiers using more than 4 lookahead.

My thought (which I've not tested) was do we need to use lookahead for the modifiers? Could the tokens for modifiers be consumed and converted to a `Modifiers` bitflags type, rather than having to rewind to re-lex them? But I don't understand the parser well at all, so I'm not sure if that's workable.

I suspect that the lookahead `VecDeque` is the biggest bottleneck in the lexer now, so it'd be great if we can find a way to get rid of it.